### ISSUE TITLE
Global settings Sharing Options appear not to be respected

### ISSUE THREAD

**Issue by sirweazel** [nextcloud/spreed#3349]
## Steps to reproduce
1. Go to settings>sharing>exclude groups from sharing>add group
2. User member of only one group (the group just excluded from sharing above) launches talk
3. User can now still create "public" conversations with share link

### Expected behaviour
I would expect all sharing features to be disabled, including within the talk app. 

### Actual behaviour
User can now create public conversations and share out a link

### Reason for filing, or feature request
I'm submitting this because i searched, and found a couple possible related issues, but from reading i couldn't tell if they are actually what i want or am descibing.  I have users i do not want to be able to have public access or share with people outside of nextcloud instance.  Example: family members/kids that i don't want to share items or "talk" with people outside of my nextcloud instance. Or in a company setting, i would want some members to have the feature to share ouside the company, but maybe most i would want this feature restricted.  It seams without honoring the global sharing settings, that talk can be used to work around the setting.  Or maybe add this setting into the talk settings.  I just assumed that maybe talk used the sharing api? 

## Talk app
7.0.2

## Server configuration
Nextcloud 17.0.5
If you need more information, i can provide it.  Currently running via the snap package server install.

**Comment by nickvergessen** [comment]
When you update to Nextcloud 18 + Talk 8, there is an option to prevent people from creating group and public conversations. That should very much cover the usecases you describe. The sharing settings only half respected in Talk, as they mostly make Talk useless quit easily. If you would not be allowed to "share" we basically would have to stop you from doing anything in Talk, as writing a chat message is also sharing information. Also in the future you can upload files to conversations which would allow you to share files to people you are not allowed to share to, etc.

**Comment by sirweazel** [comment]
@nickvergessen 
Thank you for the response. I've updated. I've spent the last couple hours trying to find the optin you mention to prevent people from creating group and public conversations, but i could not find it.  The only option (which was already there in version 7) was the "limit to groups", but this disables talk all together for the user. When you do this, they can't even log into a public link created by someone else without first logging out of nextcloud.  It could just be a config that i'm missing.  I even tried the "limit to groups" combined with the "circles" app.  What am i missing?  I only want some people to use the public share link in talk.
